### DISCUSSION TITLE
Manual de como entregar nuestras practicas

### DISCUSSION THREAD

**Discussion by AlvaroFierro** [Launch-X-Latam/MisionFrontEnd#247] [Show and tell]
He visto que varios no saben como poder entregar las practicas que vamos a ir haciendo a lo largo de nuestra trayectoria aquí en launch-x. Por lo que quiero ayudarles para que no se esté repitiendo demasiado en los canales. 

Pueden ir subiendo lo que hacemos en clase, o solo las puras practicas (lo mas recomendable). Porque vamos a rellenar un formulario donde iran nuestras practicas, asi que mientras nos mandan el link, podemos ir haciendo nuestra parte y despejar nuestras dudas.

Tambien deben de seguir el repo de [mision frontend][mision-frontend] porque ahi en la carpeta practicas de cada modulo van diciendo la fecha de entrega 👀

# Prerequisitos 
- **Git**
-  **[Curso basico de Git][git-onboarding] del Onboarding**
 
Si aun no instalas Git, [descargalo e instalalo desde la pagina oficial][git-download]

# Requisitos 
⚠️ ⚠️ **Seguir todos los pasos como se indica, de otra manera no funciona <br/> Asesgurate de leer bien y escribir bien lo que pones en la terminal** ⚠️ ⚠️ 

## Manera de entregar del repositorio

Se pueden entregar de dos maneras:
- Haciendo un repositorio general donde vayan a ir todas sus prácticas
- Haciendo un repositorio para cada práctica (no es muy recomendable, porque tendrías muchos repositorios)

Recuerden que esta formación en 100% autodidacta, de hecho estaba en los requerimientos de la convocatoria. Y es algo que nos debe de identificar, el poder resolver problemas o tareas que no sabemos mediante la búsqueda. 

# Repositorio local
## Inicializar nuestro repositorio
Pero para inicializar un repositorio sería así

- Abrir una terminal (en caso de Windows `Git Bash`)
- Ir a la carpeta donde queremos que se haga el repositorio 
- Y en nuestra terminal escibiremos lo siguiente

 ```bash
$ git init 
``` 
Esto iniciara el repositorio de manera **local** 

## Añadiendo nuestros archivos
- Despues tenemos que decirle a la computadora que archivos queremos que agregue (en este caso queremos que sean todos)
```bash
$ git add .
```

En el caso que quieras anadir un archivo en concreto, deberas de 
```bash
$ git add nombreDelArchivo.extension ...
```
## Haciendo commit
- Tendras que hacer `commit` o un mensaje que nos permita identificar los cambios que hemos hecho 
```bash
$ git commit -m "Aqui escribes como te gustaria que identifiques los cambios con el mensaje" 
```
# Repositorio en GitHub
## Creando el repositorio
Ahora tienes que subir el repositorio local a tu cuenta de github, para esto puedes hacer click en la pagina de inicio y en nuevo repositorio, o [hacer click en este enlace][new-repo]
![repo](https://user-images.githubusercontent.com/71525061/197353550-aeace967-2bd5-4e8e-a080-0a2895d66e37.png)

Despues debes rellenar los datos algo asi...
⚠️Tambien puedes modificar los datos despues, por si te equivocas o lo quieres volver a renombrar o algo asi...
![0](https://user-images.githubusercontent.com/71525061/197354201-f383539f-f3f4-49be-b867-a80f671fa6cc.png)

Dar click en el `boton verde`
Y despues te aparecera algo asi
![1](https://user-images.githubusercontent.com/71525061/197354117-9dc4e901-0e4e-4cd3-90a6-8c2355c0eb03.png)

# Enlazando tus archivos locales con GitHub

Aqui volveremos a nuestra terminal
Y escribiremos los comandos que nos dicen en la segunda parte

```bash
$ git remote add origin git@github.com:USUARIO/launch-x-latam.git
$ git branch -M main
$ git push -u origin main
```
Ahora refrescamos la pagina y nuestros archivos ahi estaran.

Espero les haya servido y si ven que demas preguntan por esto, redireccionenlos aqui ✅

[new-repo]: https://github.com/new
[mision-frontend]: https://github.com/Launch-X-Latam/MisionFrontEnd
[git-onboarding]: https://github.com/Launch-X-Latam/OnBoarding#taller-de-github
[git-download]: https://git-scm.com/downloads

**Comment by Will0220** [comment]
Muy buena ayuda!

**Comment by HoracioTorrez** [comment]
O sea que tenemos que vincular github con nuestro ordenador de alguna forma?

**Reply by AlvaroFierro** [reply]
Así es, porque en si estarás subiendo estos archivos a la nube, pero los tienes que tener de manera local

**Comment by EsroblasaLX** [comment]
Excelente ayuda, te lo agradezco mucho, ahora me queda mas claro de como hacerlo

**Comment by AndresGaliciaT** [comment]
Gracias :D

**Comment by OATS74** [comment]
gracias por el aporte , ya lo habia hecho con otras practica pero no con las que van evaluar .jjj

**Comment by monchi369** [comment]
perdon la ignorancia... como seria abrir una terminal??

**Reply by AlvaroFierro** [reply]
Si tienes Windows como sistema operativo primero tendrás que [descargar git][git] 
Y después tendras que buscar `Git Bash`
![image](https://user-images.githubusercontent.com/71525061/198898432-8c9905d0-30e0-40ed-a7a8-a13ab0c761f3.png)
Y abrirla

Si tienes Linux o MacOS deberas de buscar en tus programas la aplicacion llamada `terminal` ya que viene por defecto en sistemas operativos UNIX

[git]: https://git-scm.com/downloads

**Comment by monchi369** [comment]
![WhatsApp Image 2022-10-30 at 16 58 43](https://user-images.githubusercontent.com/115431287/198899047-1106e402-d7de-4993-b38a-8e3740e40b44.jpeg)

**Reply by monchi369** [reply]
ahi a donde tengo que ir??

**Reply by AlvaroFierro** [reply]
No, tienes que abrir `Git Bash`, no Git GUI

**Comment by Syara224** [comment]
Gracias por este excelente aporte.

**Comment by safraZX** [comment]
Pregunta, subo mis practicas a el repositorio y en los link de entrega, subo el link de mi repo, pero eso lo haría en una solo entrega o debería poner el mismo link en cada entrega ?

**Reply by AlvaroFierro** [reply]
Lee el 2do parrafo de esta discusion 👀

**Comment by SCPDeltaL049** [comment]
muchas gracias amigo, todo iba muy bien, seguí los pasos tal como pusiste pero en una indicacion me salio un error y ya no supe que mas poner, creo que ya me he atorado
Espero que alguien me pueda ayudar. Se los agradeceria bastante
.
![ERROR GIT](https://user-images.githubusercontent.com/114357907/199400296-d7e8986a-6ef9-40df-9c56-241021849449.png)

**Reply by AlvaroFierro** [reply]
No puedes hacer `git push` sin haber hecho `commit` antes de añadir los archivos para que los puedas subir 
Y de hecho para que pudieran copiar y pegar y no cometer errores puse que se pudiera copiar el texto. Solamente tienes que hacer hover sobre los comandos y del lado derecho aparecera un boton para copiar el texto

Pero si quieres, muestrame tu historial de tu terminal, solo copia y pega este comando en la `terminal`, para saber como te puedo ayudar 
```bash
$ history
```

Si te da error, quita el `$` pero lo pongo para que se den cuenta de los saltos de linea

**Reply by SCPDeltaL049** [reply]
aqui esta, perdona por responder apenas, la escuela y el trabajo me esta comiendo
![history](https://user-images.githubusercontent.com/114357907/199898380-ec95d14d-0680-4ff2-b599-b32e314699eb.jpg)

**Reply by NoirOrange** [reply]
Saludos! Si después de seguir el tutorial al pie de la letra te tira un error, escribe:
`$ git pull --rebase origin main`
Y después: 
`$ git push origin main`

Con eso debería subir tus archivos al repositorio, sin embargo si usas esos comandos recomiendo tener un repositorio para cada práctica correspondiente.
Disculpa la tardanza, espero poderte ayudar.

**Comment by monchi369** [comment]
Alvaro!! te agradezco mucho porque ayer pude subir mis prácticas gracias a tu explicación!! que andes muy bien!!

**Reply by AlvaroFierro** [reply]
Que bien @monchi369 me da gusto, para eso es la guía 🙌
Felicitaciones!!!

**Reply by DiegoIrribarren** [reply]
pues yo no entendi ajajaj me confundi mas.. volver a leer , pero no me queda claro , porq me salio un link para descargar github desktop.. eso es otra cosa?

**Reply by AlvaroFierro** [reply]
> pues yo no entendi ajajaj me confundi mas.. volver a leer , pero no me queda claro , porq me salio un link para descargar github desktop.. eso es otra cosa?

Pues entonces tienes que primero descargar `Git`, consulta los prerequisitos de esta misma discusion

**Comment by Gallardo1225** [comment]
![image](https://user-images.githubusercontent.com/114272390/199620164-a6e683a7-0ab5-4d9a-bac8-3c6dfc15ede9.png)
hasta ahi me quedo, ya no puedo segui :(

**Reply by AlvaroFierro** [reply]
Te paso lo mismo que @SCPDeltaL049 
Checa [este mismo error](https://github.com/Launch-X-Latam/MisionFrontEnd/discussions/247#discussioncomment-4033202)

**Reply by Gallardo1225** [reply]
ya logre solucionarlo gracias... solo tenía que introducir un comando forzado para que me subiera los archivos ya que por alguna razon no se me era permitido.

**Reply by NoirOrange** [reply]
@Gallardo1225 Disculpa, cómo le hiciste?? Me pasa lo mismo y no es por el error del commit.

**Comment by craonay** [comment]
Solo creamos el repositorio y subimos todo y ¿ya?,Quisiera saber como revisan eso o si hay que vincularlo de alguna manera

**Reply by NoirOrange** [reply]
Por lo que tengo entendido así las van a "revisar". Subiendo todas tus prácticas a tu repositorio
